tests/e2e-demos/e2e-demo.go

```diff
@@ -185,40 +183,118 @@ var _ = framework.E2ESuiteDescribe(Label("e2e-demo"), func() {
+
+					// get container image in case of git source
+					if componentTest.ContainerSource == "" {
 						// snapshotName is sent as empty since it is unknown at this stage
-						snapshot, err = fw.IntegrationController.GetApplicationSnapshot("", application.Name, namespace, component.Name)
+						snapshot, err := fw.IntegrationController.GetApplicationSnapshot("", application.Name, namespace, component.Name)
 						Expect(err).ShouldNot(HaveOccurred())
-					})
-				}
+						containerImage = getContainerImage(component, snapshot)
+					}
 
-				// Create snapshotEnvironmentBinding to cause an application (and its components) to be deployed
-				It("snapshotEnvironmentBinding is created", func() {
-					snapshotEnvBindingName := SnapshotEnvironmentBindingName + "-" + util.GenerateRandomString(4)
-					_, err = fw.HasController.CreateSnapshotEnvironmentBinding(snapshotEnvBindingName, namespace, application.Name, snapshot.Name, EnvironmentName, component)
-					Expect(err).NotTo(HaveOccurred())
+					applicationSnapshot_push, err = fw.IntegrationController.CreateApplicationSnapshot(application.Name, namespace, component.Name, containerImage)
+					Expect(err).ShouldNot(HaveOccurred())
+					GinkgoWriter.Printf("snapshot %s is found", applicationSnapshot_push.Name)
+				})
+
+				It("finds the snapshot with two components", func() {
+					if len(appTest.Components) > 1 &&
+						appTest.Components[len(appTest.Components)-1].Name == component.Name {
+						timeout = time.Second * 600
+						interval = time.Second * 10
+						Eventually(func() bool {
+							snap, err := fw.IntegrationController.GetApplicationSnapshot("", application.Name, namespace, "")
+							Expect(err).ShouldNot(HaveOccurred())
+							if snap != nil {
+								GinkgoWriter.Printf("Found snapshot %s with two components\n", snap.Name)
+								applicationSnapshot_push = snap
+								return true
+							}
+							return false
+						}, timeout, interval).Should(BeTrue(), "time out when trying to find the snapshot with two components")
+					}
+				})
+
+				It("checks if the global candidate is updated after push event", func() {
+					timeout = time.Second * 600
+					interval = time.Second * 10
+					Eventually(func() bool {
+						if fw.IntegrationController.HaveHACBSTestsSucceeded(applicationSnapshot_push) {
+							component, err := fw.IntegrationController.GetComponent(application.Name, namespace)
+							Expect(err).ShouldNot(HaveOccurred())
+							Expect(component.Spec.ContainerImage != "").To(BeTrue())
+							GinkgoWriter.Printf("Global candidate is updated\n")
+							return true
+						}
+
+						applicationSnapshot_push, err = fw.IntegrationController.GetApplicationSnapshot(applicationSnapshot_push.Name, "", namespace, "")
+						Expect(err).ShouldNot(HaveOccurred())
+						return false
+					}, timeout, interval).Should(BeTrue(), "time out when waiting for updating the global candidate")
+				})
+
+				It("checks if a release is created successfully", func() {
```

Comment:
Same comment as above. Not sure if we need this? @jinqi7 @dirgim

Reply:
Yep, same as above, it probably isn't required.